src/factorization.jl

```diff
@@ -38,7 +38,7 @@ end
 struct UMFPACKFactorization <: AbstractFactorization
```

Comment:
can we add a bool for whether to redo the symbolic factorization?

Reply:
Default to no redoing it.

Reply:
Where do you want that? In the algorithm? So ` UMFPACKFactorization(reuse_symbolic=false)`?